src/custom/connectors/index.ts

```diff
@@ -1,16 +1,110 @@
-import { ChainId } from '@uniswap/sdk'
+import { Web3Provider } from '@ethersproject/providers'
 import { InjectedConnector } from '@web3-react/injected-connector'
+import { WalletConnectConnector } from '@web3-react/walletconnect-connector'
+import { WalletLinkConnector } from '@web3-react/walletlink-connector'
+import { PortisConnector } from '@web3-react/portis-connector'
 
-export {
-  NETWORK_CHAIN_ID,
-  fortmatic,
-  portis,
-  network,
-  walletconnect,
-  walletlink,
-  getNetworkLibrary
-} from '@src/connectors'
-
-export const injected = new InjectedConnector({
-  supportedChainIds: [ChainId.MAINNET, ChainId.ROPSTEN, ChainId.RINKEBY, ChainId.GÖRLI, ChainId.KOVAN, ChainId.XDAI]
+import { FortmaticConnector } from 'connectors/Fortmatic'
+import { NetworkConnector } from 'connectors/NetworkConnector'
+
+const SUPPORTED_CHAIN_IDS = process.env.REACT_APP_SUPPORTED_CHAIN_IDS
+const DEFAULT_CHAIN_ID = Number(process.env.REACT_APP_CHAIN_ID || '1')
+
+const FORMATIC_KEY = process.env.REACT_APP_FORTMATIC_KEY
+const PORTIS_ID = process.env.REACT_APP_PORTIS_ID
+
+export const NETWORK_CHAIN_ID: number = parseInt(process.env.REACT_APP_CHAIN_ID ?? '1')
+
+type RpcNetworks = { [chainId: number]: string }
+
+function getRpcNetworks(): [RpcNetworks, number[]] {
+  // Make sure the mandatory envs are present
+  if (typeof SUPPORTED_CHAIN_IDS === 'undefined') {
+    throw new Error(`REACT_APP_NETWORK_URL must be a defined environment variable`)
+  }
+
+  // Get list of supported chains
+  const chainIds = SUPPORTED_CHAIN_IDS.split(',').map(chainId => Number(chainId.trim()))
+  if (chainIds.length === 0) {
+    throw new Error(`At least one network should be supported. REACT_APP_CHAIN_ID`)
+  }
+
+  // Make sure the default chain is in the list of supported chains
+  if (!chainIds.includes(DEFAULT_CHAIN_ID)) {
+    throw new Error(
+      `The default chain id (${DEFAULT_CHAIN_ID}) must be part of the list of supported networks: ${chainIds.join(
+        ', '
+      )}`
+    )
+  }
+
+  // Return rpc urls per network
+  const rpcNetworks = chainIds.reduce<RpcNetworks>((acc, chainId) => {
+    const url = process.env['REACT_APP_NETWORK_URL_' + chainId]
+
+    if (typeof url === 'undefined') {
+      throw new Error(
+        `Network ${chainId} is supported, however 'REACT_APP_NETWORK_URL_${chainId} environment variable was not defined`
+      )
+    }
+
+    acc[chainId] = url
+
+    return acc
+  }, {})
+
+  // Get chainIds (excluding the DEFAULT_CHAIN_ID)
+  // Reason: By convention we will return DEFAULT_CHAIN_ID as the first element in the supported networks
+  const otherChainIds = Object.keys(rpcNetworks)
+    .map(Number)
+    .filter(networkId => networkId !== DEFAULT_CHAIN_ID)
```

Comment:
I see that this could be more explicit but if you're already sure the `DEFAULT_CHAIN_ID` exists before the `reduce` on `L42` you could just add the `DEFAULT_CHAIN_ID` check/filter inside the `reduce` like:
```ts
...
if (chainId === DEFAULT_CHAIN_ID) return acc
...
```

Reply:
is a good advice, although I would need to add some repetition for this relevant part:

```ts
const url = process.env['REACT_APP_NETWORK_URL_' + chainId]

    if (!url) {
      throw new Error(
        `Network ${chainId} is supported, however 'REACT_APP_NETWORK_URL_${chainId} environment variable was not defined`
      )
    }
```